addons/event/models/event_registration.py

```diff
@@ -139,22 +126,25 @@ def create(self, vals_list):
             # we currently avoid this by not running the scheduler, would be best to find the actual
             # reason for this issue and fix it so we can remove this check
             registrations._update_mail_schedulers()
-
         return registrations
 
     def write(self, vals):
-        pre_draft = self.env['event.registration']
-        if vals.get('state') == 'open':
-            pre_draft = self.filtered(lambda registration: registration.state == 'draft')
-
+        confirming = vals.get('state') == 'open'
+        pre_draft = (self.filtered(lambda registration: registration.state == 'draft')
+                     if confirming else None)
         ret = super(EventRegistration, self).write(vals)
-
-        if vals.get('state') == 'open' and not self.env.context.get('install_mode', False):
-            # running the scheduler for demo data can cause an issue where wkhtmltopdf runs during
-            # server start and hangs indefinitely, leading to serious crashes
-            # we currently avoid this by not running the scheduler, would be best to find the actual
-            # reason for this issue and fix it so we can remove this check
-            pre_draft._update_mail_schedulers()
+        # Necessary triggers as changing registration states cannot be used as triggers for the
```

Comment:
When creating a registration, the constraint on event and ticket are called, isn't it, hence the override on write ? In that case would complete the explanation. If not, well ...